### PR TITLE
Sweep rework

### PR DESCRIPTION
**Pull request by dorijanko**
This PR changes the resource state from a Counter to an Int, ignores Core's sweep and makes the same entrance with the same possible resource states not try to use the same state modifier multiple times, along with a few minor bugfixes(most notably, I'm pretty sure creating playthrough shouldn't fail anymore). From my testing these changes seem to speed up generation by a factor of about 2x.

### CODE REVIEW

File: worlds/hk/resource_state_vars/soul_manager.py [hunk 1]
```diff
@@ -82,59 +85,57 @@ def try_resore_soul(self):
...
+            state_blob = rs_set_value(state_blob, "SPENTRESERVESOUL", 0)
+        return True, state_blob
 
     def get_soul_info(self, state_blob: rs, item_state: cs) -> SoulInfo:
-        max_soul = 99 - state_blob["SOULLIMITER"]
-        soul = max_soul - state_blob["SPENTSOUL"]
-        vessels = item_state.count("VESSELFRAGMENTS", self.player) // 3
+        max_soul = 99 - rs_get_value(state_blob, "SOULLIMITER")
+        soul = max_soul - rs_get_value(state_blob, "SPENTSOUL")
+        vessels = min(6, item_state.count("VESSELFRAGMENTS", self.player) // 3) # max 6 so that state doesn't overflow
```
Comment (qwint) [inline worlds/hk/resource_state_vars/soul_manager.py]: can you add tests to validate for this state_blob implementation these attributes don't overflow?
they can be very targeted to the functions that we have special cases for but want coverage into these cases for the future

**Comment by qwint** [comment]
Merged into #8 and released in 1.1.0
thanks for the contribution!
